pkg/mcs/tso/server/apis/v1/api.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apis
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gin-contrib/cors"
+	"github.com/gin-contrib/gzip"
+	"github.com/gin-gonic/gin"
+	tsoserver "github.com/tikv/pd/pkg/mcs/tso/server"
+	"github.com/tikv/pd/pkg/utils/apiutil"
+	"github.com/tikv/pd/server"
+	"github.com/unrolled/render"
+)
+
+// APIPathPrefix is the prefix of the API path.
+const APIPathPrefix = "/"
+
+var (
+	apiServiceGroup = apiutil.APIServiceGroup{
+		Name:       "tso",
+		Version:    "v1",
+		IsCore:     false,
+		PathPrefix: APIPathPrefix,
+	}
+)
+
+func init() {
+	tsoserver.SetUpRestHandler = func(srv *tsoserver.Service) (http.Handler, apiutil.APIServiceGroup) {
+		s := NewService(srv)
+		return s.handler(), apiServiceGroup
+	}
+}
+
+// Service is the resource group service.
+type Service struct {
+	apiHandlerEngine *gin.Engine
+	baseEndpoint     *gin.RouterGroup
+
+	srv *tsoserver.Service
+	rd  *render.Render
+}
+
+func createIndentRender() *render.Render {
+	return render.New(render.Options{
+		IndentJSON: true,
+	})
+}
+
+// NewService returns a new Service.
+func NewService(srv *tsoserver.Service) *Service {
+	apiHandlerEngine := gin.New()
+	apiHandlerEngine.Use(gin.Recovery())
+	apiHandlerEngine.Use(cors.Default())
+	apiHandlerEngine.Use(gzip.Gzip(gzip.DefaultCompression))
```

Comment:
Maybe need the `ServiceRedirector` middleware like https://github.com/tikv/pd/issues/6051.

Reply:
OK. Thanks a lot :-)